src/Workspaces/CSharp/Portable/LanguageServices/CSharpSyntaxFactsService.cs

```diff
@@ -246,6 +246,32 @@ public bool IsForEachStatement(SyntaxNode node)
             return node is ForEachStatementSyntax;
         }
 
+        public bool IsDeconstructionForEachStatement(SyntaxNode node)
+        {
+            return node is ForEachVariableStatementSyntax;
+        }
+
+        public bool IsDeconstructionAssignment(SyntaxNode node)
+        {
+            return node is AssignmentExpressionSyntax assignment && assignment.IsDeconstruction();
+        }
+
+        public Location GetDeconstructionReferenceLocation(SyntaxNode node)
+        {
+            var tree = node.SyntaxTree;
+            switch (node)
+            {
+                case AssignmentExpressionSyntax assignment:
+                    return Location.Create(tree, TextSpan.FromBounds(assignment.SpanStart, assignment.OperatorToken.Span.End));
```

Comment:
I am... barely ok with this :D    i think it would be better to just select hte =, and the 'in' token though.  But i would be open for arguments otherwise.